### DISCUSSION TITLE
cloudflare的数据库和对象存储会被恶意刷破产吗

### DISCUSSION THREAD

**Discussion by xiyuvi** [OXeu/Rin#159] [Q&A] [answered]
请教一下各位，cloudflare的数据库和对象存储会被恶意刷破产吗

**Comment by OXeu** [comment]
R2 对象存储参见 https://github.com/OXeu/Rin/issues/147
数据库每日有免费额度，只要你不开通付费订阅每日额度用尽以后当日就无法访问了，不会产生额外计费

**Reply by xiyuvi** [reply]
> R2 对象存储参见 #147 数据库每日有免费额度，只要你不开通付费订阅每日额度用尽以后当日就无法访问了，不会产生额外计费

感谢回复，谢谢你开发出这么棒的博客系统。我不知道这样可行不，直接用Workers新增或修改github仓库的md文件和图片，这样是不是就不用使用D1和 R2了

**Reply by OXeu** [reply]
> > R2 对象存储参见 #147 数据库每日有免费额度，只要你不开通付费订阅每日额度用尽以后当日就无法访问了，不会产生额外计费
> 
> 感谢回复，谢谢你开发出这么棒的博客系统。我不知道这样可行不，直接用Workers新增或修改github仓库的md文件和图片，这样是不是就不用使用D1和 R2了

这个思路也是可以的，但是本项目目前肯定不会这么做了

**Reply by kakomonn** [reply]
可否考虑让 workers 去读 r2 的静态资源，在这种情况下，workers 额度用完了 r2 就读不到了，也避免被恶意攻击

**Comment by kafuneri** [comment, marked as answer]
vps论坛有人把R2地址发出来让测试，一晚上被刷了1.3亿次请求 ，6TB流量。免费额度差不多够跑400GB流量，CF的R2还能套cdn缓存，只要不是故意让人来测试，应该没人攻击，攻击的会先破产吧，毕竟1千万次请求就0.36刀。

**Reply by kakomonn** [reply]
如果刷请求收费，随机路径请求返回404错误也会被计算B类收费
每次请求都是不同随机路径，会导致无法缓存必回源，而且只请求头，不占攻击者带宽
![image](https://github.com/user-attachments/assets/cf2cf531-12f2-44b8-9363-5d66ad7a96df)
